### PR TITLE
feat/AUT-76/stylesheet-manager-for-passages

### PR DESCRIPTION
**Pull request by marpesia**
**Related to:** [AUT-76](https://oat-sa.atlassian.net/browse/AUT-76)

**Description:**
- Add a Stylesheet manager for passages and isolate the scope of the CSS rules to the passage.
- The stylesheets attached to a passage should not affect the style of the items the passage is included into

**Changes:**

- Refactor handle functions between taoMediaManager and taoItemQti

:exclamation: Pending Solar Preview, depends on Solar Delivery part :exclamation:

**PR related:**
- https://github.com/oat-sa/extension-tao-mediamanager/pull/404

**How to check:**

- Check Items and Test with passages with stylesheets.
- Check adding passage with custom styles and without it and CSS file. (Without custom styles may have not mainClass on the passage)
- Check previews Terre.

[Kitchen environment on the ticket.](https://oat-sa.atlassian.net/browse/AUT-76?focusedCommentId=190406)

### CODE REVIEW

**Review by bziondik — COMMENTED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

File: views/js/qtiCreator/helper/xincludeRenderer.js [hunk 1]
```diff
@@ -64,8 +72,7 @@ define([
                         //reload the wiget to rfresh the rendering with the new href
                         xincludeWidget.refresh();
                     }, loadedClasses);
-
-                    _.each(xincludeHandlers, handler => handler(xinclude.attr('href')));
+                    _.each(xincludeHandlers, handler => handler(xinclude.attr('href'), className, xi.serial));
```
Comment (meefox) [inline views/js/qtiCreator/helper/xincludeRenderer.js:75]: **suggestion**: we can replace underscore code with native code as well in the future conversations to ES6

File: views/package.json [hunk 2]
```diff
@@ -9,6 +9,6 @@
     },
     "dependencies": {
         "@oat-sa/tao-item-runner": "0.8.1",
-        "@oat-sa/tao-item-runner-qti": "1.4.0"
+        "@oat-sa/tao-item-runner-qti": "https://github.com/oat-sa/tao-item-runner-qti-fe#feat/AUT-76/stylesheet-manager-for-passages"
```
Comment (meefox) [inline views/package.json:12]: - [ ] just as a reminder to update dependency to released version

**Review by meefox — APPROVED** [review]
Well done and good job moving to ES6!

**Review by meefox — APPROVED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: views/js/qtiCreator/editor/styleEditor/styleSheetToggler.js [hunk 3]
```diff
@@ -33,16 +33,35 @@ define([
...
+
+                $('body').append($messageBox);
+
+                closeTrigger.on('click', function () {
+                    $messageBox.fadeOut(function () {
+                        $(this).remove();
+                    });
+                });
+
+                setTimeout(function () {
```
Comment (bziondik) [inline views/js/qtiCreator/editor/styleEditor/styleSheetToggler.js:48]: Why we need SetTimeout in 4523 milliseconds?

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

image

File: views/js/qtiCreator/editor/styleEditor/styleSheetToggler.js (hunk 3 above)
Comment (marpesia) [reply views/js/qtiCreator/editor/styleEditor/styleSheetToggler.js:48]: This setTimeout was already there, and I just moved the function before it was calling. I had a linter issue with that.
I understand that it is forcing the modal to close if the user hasn't closed it after a while.

Comment (bziondik) [reply views/js/qtiCreator/editor/styleEditor/styleSheetToggler.js:48]: can you change it to more 'beautiful ' number? like 4000?

**Pull request merged**